synapse/rest/client/room.py

```diff
@@ -1284,17 +1284,14 @@ class TimestampLookupRestServlet(RestServlet):
     `dir` can be `f` or `b` to indicate forwards and backwards in time from the
     given timestamp.
 
-    GET /_matrix/client/unstable/org.matrix.msc3030/rooms/<roomID>/timestamp_to_event?ts=<timestamp>&dir=<direction>
+    GET /_matrix/client/v1/rooms/<roomID>/timestamp_to_event?ts=<timestamp>&dir=<direction>
     {
         "event_id": ...
     }
     """
 
     PATTERNS = (
-        re.compile(
-            "^/_matrix/client/unstable/org.matrix.msc3030"
-            "/rooms/(?P<room_id>[^/]*)/timestamp_to_event$"
-        ),
+        re.compile("^/_matrix/client/v1/rooms/(?P<room_id>[^/]*)/timestamp_to_event$"),
```

Comment:
I just matched what `/threads` was doing since `client_patterns(...)` seems to be doing weird stuff with `api/v1` and `unstable` by default. Not sure why those are things for normal endpoints.

https://github.com/matrix-org/synapse/blob/a84744fba0b123eef31a179648177e9d4aacc693/synapse/rest/client/relations.py#L85

Reply:
I think we're trying to not use `client_patterns()` anymore since it hides a lot of logic.

Is anything using the unstable version that we need to care about or can we just swap right away?

Reply:
> Is anything using the unstable version that we need to care about or can we just swap right away?

I only know of the Element Web jump to date UI and the Matrix Public Archive. 

> Client implementations:
> 
>  - `/jumptodate` slash command, https://github.com/matrix-org/matrix-react-sdk/pull/7372
>  - Jump to date headers (with date picker) in timeline: https://github.com/matrix-org/matrix-react-sdk/pull/7339
>  - Matrix Public Archive: https://github.com/matrix-org/matrix-public-archive


But those things are experimental along with this endpoint and have no guarantees.

 - We can update the `matrix-js-sdk` to try to pull from both the old unstable and new place.
 - And we can just update the Matrix Public Archive with whatever

Reply:
If those are checking for the version flag than it sounds like it should be fine! 🎉